plugins/render_fortran_examples.py

```diff
@@ -0,0 +1,252 @@
+"""Render the Fortran examples from the Cantera repository into Nikola listings.
+
+This plugin renders Fortran examples from the main Cantera repository into the
+examples/fortran output folder. It looks for the examples in the folder configured
+in the top-level conf.py file in the ``EXAMPLES_FOLDERS`` dictionary. That
+dictionary has keys with the source folder and values with the destination
+folder (relative to the ``OUTPUT_FOLDER``). The relevant source folder is found
+as the key associated with the value that contains the string ``fortran``,
+typically ``"../cantera/samples/f90": "examples/fortran"``.
+"""
+from pathlib import Path
+
+from nikola.plugin_categories import Task
+from nikola import utils
+from pygments import highlight
+from pygments.lexers import FortranLexer, CppLexer
+from itertools import chain
+import re
+import natsort
+
+
+def render_example_index(site, kw, headers, output_file):
+    """Render the index of all of the Fortran examples.
+
+    Parameters
+    ==========
+    site:
+        An instance of a Nikola site, available in any plugin as ``self.site``
+    kw:
+        A dictionary of keywords for this task
+    headers:
+        A dictionary of the example categories and the summaries of each example
+    output_file:
+        A pathlib.Path instance representing the output file that will be rendered
+
+    """
+    n = 3
+    for head_dict in headers.values():
+        head_files = head_dict["files"]
+        head_dict["files"] = [
+            head_files[i : i + n] for i in range(0, len(head_files), n)  # NOQA: E203
+        ]
+
+    permalink = output_file.relative_to(kw["output_folder"])
+    title = "Fortran Examples"
+    context = {
+        "headers": headers,
+        "lang": kw["default_lang"],
+        "pagekind": ["example"],
+        "permalink": str(permalink),
+        "title": title,
+        "description": title,
+    }
+    site.render_template("fortran-example-index.tmpl", str(output_file), context)
+
+
+def render_example(site, kw, in_name, out_name):
+    """Render a single source file to HTML with formatting.
+
+    Parameters
+    ==========
+    site:
+        An instance of a Nikola site, available in any plugin as ``self.site``
+    kw:
+        A dictionary of keywords for this task
+    in_name:
+        The file to be rendered, as an instance of pathlib.Path
+    out_name:
+        A pathlib.Path instance pointing to the rendered output file
+
+    """
+    if str(in_name).endswith('.cpp'):
+        lexer = CppLexer()
+    else:
+        lexer = FortranLexer()
```

Comment:
The `FortranLexer()` appears not to like the format of our Fortran files... `c` lines aren't rendered as comments, and the `$` shows as an error. Is there a better lexer, or do we just have to fix up the examples?

Reply:
Or do we just ignore it and move on? :joy: